### DISCUSSION TITLE
How to turn off damping for TetModel

### DISCUSSION THREAD

**Discussion by Theodotus1243** [InteractiveComputerGraphics/PositionBasedDynamics#145] [Q&A] [answered]
Hi @janbender
I've set `gravity = [0, 0, 0]` and initialized `addRegularTetModel` with `addSolidConstraints`
Then I've spinned it to 200Hz, but for some reason after 2 seconds angular velocity reduced a lot

I thought it had to do with some kind of friction. But `setFrictionCoeff(0)` and `setRestitutionCoeff(1)` did not help.
Also I've found `sim.setDamping(0.001f);` but it's only related `PositionBasedElasticRodsDemo.cpp`
__________________
I think it's related to `DamperJoint` from which the object is created. But I haven't found how to zero damping coef

**Comment by Theodotus1243** [comment]
Actually there is no problem if I need to add some Joints between vertices The question is do all the Joints have damping and where it comes in Tet variant

**Comment by janbender** [comment, marked as answer]
Which method did you choose in the call of addSolidConstraints? Friction and restitution only applies to collisions. So if there are no collisions, this won't change anything. 

Note that PBD suffers from numerical damping (similar to the implicit Euler integrator). You can improve this by using the second-order velocity update. Otherwise you can test if it gets better when reducing the time step size.

**Reply by Theodotus1243** [reply]
Thank You very much
I was using `simMethod = 3`, but tried `1-6` too, paying more attention at `1,3,6`, no principal difference. I've looked and them and saw that `simMethod = 1` is just distance and volume constraint with no explicit damping, so thought that it's numerical damping 

Time step reduction in 10-100 times looks like further accererates damping. Making object bigger slows down damping
(Not sure about this two statements, but those changes don't help much)

Hoping a lot on second-order velocity update

**Comment by Theodotus1243** [comment]
@janbender Thanks
I've set 
```
ts = sim.getTimeStep()
ts.setValueInt(
    pbd.TimeStepController.VELOCITY_UPDATE_METHOD, 
    pbd.TimeStepController.ENUM_VUPDATE_SECOND_ORDER
)
```

And it's looks much better
The issue lookes resolved for now
